src/main/scala/com/workday/warp/persistence/mysql/WarpSlickFunctions.scala

```diff
@@ -0,0 +1,26 @@
+package com.workday.warp.persistence.mysql
+
+import com.workday.warp.persistence.mysql.WarpSlickFunctions._
+import slick.ast.Library.SqlAggregateFunction
+import slick.ast.TypedType
+import slick.lifted.FunctionSymbolExtensionMethods._
+import slick.lifted.{Query, Rep}
+import scala.language.{higherKinds, implicitConversions}
+
+/**
+  * Created by ruiqi.wang
+  *
+  */
+protected class WarpSlickFunctions[B1, P1, C[_]](val q: Query[Rep[P1], _, C]) extends AnyVal {
+  type OptionTM = TypedType[Option[B1]]
```

Comment:
😕

Reply:
See here: 

https://github.com/slick/slick/blob/master/slick/src/main/scala/slick/ast/Type.scala#L253

https://github.com/slick/slick/blob/master/slick/src/main/scala/slick/ast/Type.scala#L337

https://github.com/slick/slick/blob/master/slick/src/main/scala/slick/ast/Library.scala#L102

https://github.com/slick/slick/blob/master/slick/src/main/scala/slick/lifted/FunctionSymbolExtensionMethods.scala

Essentially telling the slick compiler that the `STD` symbol has valid typing (which is used to compile the actual sql query).